src/utils/searchViewHelper.js

```diff
@@ -51,6 +54,22 @@ const searchViewHelper = {
       searchViewHelper.searchByRevisionOrEmail(repository, search);
     }
   },
+
+  handleClickOutsideInput(event) {
+    try {
+      if (
+        event.target.id === 'search-revision-input' ||
+        event.target.classList.contains('search-revision') ||
+        event.target.parentElement.classList.contains('search-revision')
+      ) {
+        store.dispatch(updateSearchIsFocused(true));
+      } else {
+        store.dispatch(updateSearchIsFocused(false));
```

Comment:
I don't have the big picture clear in my head yet, but it seems to me that having this state in redux makes things more difficult. I believe I'd keep this state inside the React component. Indeed this state isn't useful throughout the full app, but only for this specific UI component.

I feel like this comes from the fact that the event handlers are not in the React component, but instead in this separate object 'searchViewHelper` that does direct calls to `store.dispatch`.

To be explicit: I don't really mind having it in Redux for now so that this can move forward. In the future it could be good to think about components as reusable blocks that can also take props from their parent instead of taking all their props from redux.

Reply:
> It would be better to test for this specific case instead of catching the error. If for some reason you have a `TypeError` anywhere else (for example in the reducers that run when dispatching the action) this would hide it.

after using `matches` from your previous comment, it appears I no longer need to check for an error :ok_hand:

Reply:
so I was considering making these changes now, but as you stated we have separated UI from logic. I don't know what the general consensus on this is, but from my (beginner's) perspective it does seem cleaner. 

I have an easier time of comprehending everything that is going on when components are UI only and logic is handled in a separate file. 

It's worth noting that Perfherder does _not_ separate UI from logic, and this tool intends to replace it (or at least the compare view, for now). There is quite a lot of computation and passing props between components, and it is very hard to follow.

I imagine that is one reason I'm opinionated on the matter :)

Reply:
We can discuss this further later :-)

I'd just want to make sure that you know about the `mapDispatchToProps` parameter to the react-redux `connect` method: this makes it very easy to connect the action dispatches with the react components. There's not much passing props going on when working with connected components.

I think my biggest concern is how `store` is like a global that any file can import anywhere and dispatch actions to. Normally the `store` object is only accessed indirectly through `react-redux`'s `connect` function using the `mapDispatchToProps` option or the `dispach` prop. (Or when using the hooks API, with `useDispatch`).
But obviously with these "helper" files it would mean passing the `dispatch` function to all functions that are called, and  understand this would be cumbersome.

Anyway this can be work for later, I think it would be fairly easy to move all code from the "helper" files back to the same files as the react components, so there's nothing to be really worried about here.